### PR TITLE
Add QR identifier on check-in

### PR DESCRIPTION
**Pull request by oriolclosa**
## What this PR does / Why we need it
Adds QR identification for next HackUPC edition. Adds `qr_identifier` field on `CheckIn` model, updates the QR scanner in order to work on laptops and adds the assignment of the code on check-in.

## Some questions
- [X] I have read the contributing guidelines
- [X] I abide by this repository Code of Conduct
- [X] I understand that my PR won't be merged until Travis gives a "green light"

### CODE REVIEW

**Comment by casassg** [comment]
Looks good. I would probably make the QR code to redirect to the checkin page for that hacker directly (instead of just filling the Search field with the QR content)

File: checkin/static/js/checkin.js [hunk 1]
```diff
@@ -0,0 +1,114 @@
...
+        let selectCam = document.createElement("select")
+        let optionsStr=""
+        for(let i =0; i < cams.length; i++)
+            optionsStr += "<option value='"+i+"'>" + (cams[i].name || "Camera "+i) + "</option>"
+        selectCam.innerHTML=optionsStr
+        popup.appendChild(selectCam)
+        selectCam.value = ""+selectedCam
+        //On selector change, we stop the scanner preview and change the camera
+        selectCam.addEventListener("change", ()=>{
+            let selectedCam = parseInt(this.value)
```
Comment (casassg) [inline checkin/static/js/checkin.js]: This fails for some Android Phones. this.value is not always set to the correct value.

Comment (casassg) [reply checkin/static/js/checkin.js]: Fixed this same problem in meals

File: checkin/views.py [hunk 2]
```diff
@@ -64,11 +57,16 @@ def get_context_data(self, **kwargs):
 
     def post(self, request, *args, **kwargs):
         appid = request.POST.get('app_id')
+        qrcode = request.POST.get('qr_code')
+        if qrcode is None or qrcode == "":
```
[the remark this answers is not in the document]
Comment (oriolclosa) [reply checkin/views.py]: Yes, changed to single ones.

File: checkin/tables.py [hunk 3]
```diff
@@ -27,7 +28,7 @@ class Meta:
         attrs = {'class': 'table table-hover'}
         template = 'django_tables2/bootstrap-responsive.html'
         fields = ['user.name', 'user.email']
-        empty_text = 'All hackers checked in! Yay!'
+        empty_text = 'No hacker found, are them all checked-in?'
```
[the remark this answers is not in the document]
Comment (oriolclosa) [reply checkin/tables.py]: Changed.

**Comment by oriolclosa** [comment]
Hmm, yes, will do the redirect only if one instance was found, right?

File: checkin/static/js/checkin.js (hunk 1 above)
Comment (oriolclosa) [reply checkin/static/js/checkin.js]: Fixed using your correction from `meals`.

**Comment by oriolclosa** [comment]
Won't do, the only way I found without changing the FilterView was to update the context and then on the template redirect with a meta tag if context contained something set by me previously on the update context file in case the count of applications filtered by search was 1. This duplicated the search also...

**Comment by oriolclosa** [comment]
If there's an easy way, I couldn't find it...

**Comment by casassg** [comment]
Make it easier. On QR detected callback just do:

```
location.href = 'checkin/' + qrcode
```

Not the cleanest, but works nice

**Comment by casassg** [comment]
That's how the existing QR code reader works at the moment.

https://github.com/hackupc/registration/blob/master/checkin/static/webqr/webqr.js#L55

**Comment by oriolclosa** [comment]
Added the redirect after checking if it's an UUID, in case it's not, it just filters the results. Although this would cause a 404 in case the UUID is not of any application, it's the maximum we can do here.

**Pull request merged**